common-money.lic

```diff
@@ -64,11 +99,74 @@ module DRCM
   def check_wealth(currency)
     DRC.bput("wealth #{currency}", /\(\d+ copper #{currency}\)/i, /No #{currency}/i).scan(/\d+/).first.to_i
   end
-
+  
   def wealth(hometown)
     check_wealth(hometown_currency(hometown))
   end
 
+  def get_total_wealth
+    # This method captures your current total on-hand wealth
+    # and returns a hash representing the numerical value in
+    # coppers of each currency.    
+
+    # Set up variables to capture each line of wealth output by currency
+    kronars_line = nil
+    lirums_line = nil
+    dokoras_line = nil
+
+    # Set up variables to capture the value in coppers of each currency
+    # Set to zero so that, if we have, for example, "No Lirums"
+    # we simply return the initialized value of 0.
+    kronars = 0
+    lirums = 0
+    dokoras = 0
+
+    # Grab the character's wealth, pausing a bit
+    # then grabbing a sufficient number of lines
+    # to ensure we get all the output taking into
+    # account other random scroll text.
+    # Reversing the lines ensures we are processing
+    # the most recent output from 'wealth', in case
+    # reget were to grab output from back-to-back calls.
+    DRC.bput("wealth", "Wealth")
+    pause 0.5
+    wealth_lines = reget(10).map(&:strip).reverse
+
+    # We've reversed the reget array. Now we'll iterate over it and capture
+    # each line after we recognize we've hit the Wealth block.
+    wealth_lines.each do |line|
+      case line
+      when  /^Wealth:/i
+        # This is the start of our Wealth lines.
+        # We don't need to parse this line. Break out of loop.
+        break
+      when /\(\d+ copper Kronars\)|^No Kronars/i
+        kronars_line = line
+      when /\(\d+ copper Lirums\)|^No Lirums/i
+        lirums_line = line
+      when /\(\d+ copper Dokoras\)|^No Dokoras/i
+        dokoras_line = line
+      end
+    end
+
+    # Scan each wealth line for the value in coppers,
+    # capture that numeric value, then integerize.
+    # If the line indicates no wealth of that currency, accept
+    # the instantiated value of 0 for that currency.
+    kronars = kronars_line.scan(/\((\d+) copper Kronars\)/i).first.first.to_i unless kronars_line =~ /^No/    
+    lirums = lirums_line.scan(/\((\d+) copper lirums\)/i).first.first.to_i unless lirums_line =~ /^No/        
+    dokoras = dokoras_line.scan(/\((\d+) copper dokoras\)/i).first.first.to_i unless dokoras_line =~ /^No/
```

Comment:
💡 Since the parsing of amount out of the wealth lines is contextual to the denomination, we can consolidate the logic to occur in the loop that identifies the wealth lines, thus removing some duplicated regex patterns.

```suggestion
    wealth_lines.each do |line|
      case line
      when /^Wealth:/i
        # This is the start of our Wealth lines.
        # We don't need to parse this line. Break out of loop.
        break
      when /\(\d+ copper Kronars\)/i
        kronars = line.scan(\d+).first.to_i
      when /\(\d+ copper Lirums\)/i
        lirums = line.scan(\d+).first.to_i
      when /\(\d+ copper Dokoras\)/i
        dokoras = line.scan(\d+).first.to_i
      end
    end
```

Additionally, with the above you can then get rid of the `<denomination>_line` variables as they aren't used anymore.